pkg/handler/csv/loki_csv.go

```diff
@@ -26,9 +26,13 @@ func GetCSVData(qr *model.QueryResponse, columns []string) ([][]string, error) {
 func manageStreams(streams model.Streams, columns []string) ([][]string, error) {
 	//make csv datas containing header as first line + rows
 	datas := make([][]string, 1)
+	//prepare columns for faster lookup
+	columnsMap := utils.GetMapInterface(columns)
 	//set Timestamp as first data
-	if columns == nil || utils.Contains(columns, timestampCol) {
+	includeTimestamp := false
+	if _, exists := columnsMap[timestampCol]; columns == nil || exists {
```

Comment:
Could it happen that `columns != nil` but `len(columns)==0`?

Reply:
I don't know but I can write `len(columns) == 0` instead of `columns == nil` for safety